Stack/Opc.Ua.Core/Security/Certificates/CertificateValidator.cs

```diff
@@ -691,8 +692,60 @@ private void SaveCertificates(X509Certificate2Collection certificateChain)
                         ICertificateStore store = m_rejectedCertificateStore.OpenStore();
                         try
                         {
+                            var certs = store.Enumerate().GetAwaiter().GetResult();
+                            int existingCertsCount = certs.Count;
```

Comment:
that is a very expensive operation...

Reply:
I think better were a new method in the ICertificateStore interface which can be optimized per store, this generic implementation could become an issue due to the overhead to load and decode all certs from the store.

Reply:
There will be max kMaxRejectedCerts in the rejected store normally